src/sensors/icm20948sensor.cpp

```diff
@@ -34,204 +34,93 @@ int bias_save_periods[] = { 120, 180, 300, 600, 600 }; // 2min + 3min + 5min + 1
 // Accel scale conversion steps: LSB/G -> G -> m/s^2
 constexpr float ASCALE_4G = ((32768. / ACCEL_SENSITIVITY_4G) / 32768.) * EARTH_GRAVITY;
 
-// #ifndef ENABLE_TAP
-//     #define ENABLE_TAP false
-// #endif
-
-void ICM20948Sensor::save_bias(bool repeat) {
-#if defined(SAVE_BIAS) && SAVE_BIAS
-    #ifdef DEBUG_SENSOR
-    m_Logger.trace("Saving Bias");
-    #endif
+void ICM20948Sensor::motionSetup() 
+{
+    connectSensor();
+    startDMP();
+    load_bias();
+    startMotionLoop();
+    startCalibrationAutoSave();
+}
 
-    imu.GetBiasGyroX(&m_Calibration.G[0]);
-    imu.GetBiasGyroY(&m_Calibration.G[1]);
-    imu.GetBiasGyroZ(&m_Calibration.G[2]);
+void ICM20948Sensor::motionLoop() 
+{
+#if ENABLE_INSPECTION
+    {
+        (void)imu.getAGMT();
 
-    imu.GetBiasAccelX(&m_Calibration.A[0]);
-    imu.GetBiasAccelY(&m_Calibration.A[1]);
-    imu.GetBiasAccelZ(&m_Calibration.A[2]);
+        float rX = imu.gyrX();
+        float rY = imu.gyrY();
+        float rZ = imu.gyrZ();
 
-    #if !USE_6_AXIS
-    imu.GetBiasCPassX(&m_Calibration.C[0]);
-    imu.GetBiasCPassY(&m_Calibration.C[1]);
-    imu.GetBiasCPassZ(&m_Calibration.C[2]);
-    #endif
+        float aX = imu.accX();
+        float aY = imu.accY();
+        float aZ = imu.accZ();
 
-    #ifdef DEBUG_SENSOR
-    m_Logger.trace("Gyrometer bias:     [%d, %d, %d]", UNPACK_VECTOR_ARRAY(m_Calibration.G));
-    m_Logger.trace("Accelerometer bias: [%d, %d, %d]", UNPACK_VECTOR_ARRAY(m_Calibration.A));
-        #if !USE_6_AXIS
-    m_Logger.trace("Compass bias:       [%d, %d, %d]", UNPACK_VECTOR_ARRAY(m_Calibration.C));
-        #endif
-    #endif
+        float mX = imu.magX();
+        float mY = imu.magY();
+        float mZ = imu.magZ();
 
-    SlimeVR::Configuration::CalibrationConfig calibration;
-    calibration.type = SlimeVR::Configuration::CalibrationConfigType::ICM20948;
-    calibration.data.icm20948 = m_Calibration;
-    configuration.setCalibration(sensorId, calibration);
-    configuration.save();
+        Network::sendInspectionRawIMUData(sensorId, rX, rY, rZ, 255, aX, aY, aZ, 255, mX, mY, mZ, 255);
+    }
+#endif
 
-    if (repeat) {
-        bias_save_counter++;
-        // Possible: Could make it repeat the final timer value if any of the biases are still 0. Save strategy could be improved.
-        if (sizeof(bias_save_periods) != bias_save_counter) {
-            timer.in(
-                bias_save_periods[bias_save_counter] * 1000,
-                [](void* arg) -> bool {
-                    ((ICM20948Sensor*)arg)->save_bias(true);
-                    return false;
-                },
-                this);
+    timer.tick();
+    isDataToRead = true;
+    while (isDataToRead) 
```

Comment:
I would like it more if you would make a function that just returns a boolean which you check here instead of using a class member.